geode-core/src/main/java/org/apache/geode/management/internal/configuration/domain/Configuration.java

```diff
@@ -164,9 +165,10 @@ public void toData(DataOutput out) throws IOException {
     DataSerializer.writeString(propertiesFileName, out);
     DataSerializer.writeProperties(gemfireProperties, out);
     // Before 1.12, this code wrote a non-null HashSet of jarnames to the output stream.
-    // As of 1.12, it writes a null HashSet to the stream, so that when we can still read the old
-    // configuration, and will now also write the deployment map.
-    DataSerializer.writeHashSet(null, out);
+    // As of 1.12, it will also send the hashmap with the deployments, so that the new
+    // members can read the deployments while the older members can read just the hashset
+    HashSet<String> jarNames = new HashSet<>(deployments.keySet());
+    DataSerializer.writeHashSet(jarNames, out);
```

Comment:
do you need to make a change here? Can we continue to write `null` to the stream instead of an empty set? this way, we can say, that if "jarName is not null, then it's definitely pre 1.12).

Reply:
We are not sending an empty set but rather a jarFileNames set. The problem is when older version 1.12 locators get this as `null` and it starts throwing NPE and corrupts the config